src/hooks/resources/device.ts

```diff
@@ -576,3 +589,59 @@ async function checkSupervisorReleaseUpgrades(
 		}
 	}
 }
+
+async function setSupervisorReleaseResource(
+	api: sbvrUtils.PinejsClient,
+	deviceIds: number[],
+	supervisorVersion: string,
+) {
+	// This function _should_ get called with one device (since each supervisor reports itself)
+	const device = (await api.get({
+		resource: 'device',
+		id: deviceIds[0],
+		options: {
+			$expand: ['is_of__device_type'],
```

Comment:
Can you add a $select for `is_of__device_type` please

Reply:
I'm not sure i follow, I do `$select` that in line 604: https://github.com/balena-io/open-balena-api/pull/356/files#diff-2a82ae8626a83fe22f032b4e504317d5R604

Reply:
I'm not sure I follow this, I believe it is selected in the following line?

Reply:
If you expand here, its going to pull the entire device type object, whereas doing
```
$expand: {
  is_of__device_type: {
    $select: 'id'
  }
}
```
it won't

Reply:
I also don't know what that select at the bottom is doing, I don't think you need the `is_of__device_type` there, but @Page- can confirm.

Reply:
@CameronDiver it's redundant as anything you expand is naturally also selected, but having it won't cause an issue

Reply:
Thanks for explaining that! I didn't realize i could `$select` inside my `$expand`. Updated.